pyaerocom/units/units.py

```diff
@@ -234,6 +234,7 @@ def _validate_convertible(self, other: str | Unit) -> None:
                 raise ValueError(
                     f"cfunit '{self._cfunit}' is not convertible to cfunit '{other._cfunit}'."
                 )
+            return
```

Comment:
`positive if`, then `negative if` and then a `return` statement are hard to read and very fragile.

`same_species` and `compatble_element` is calculated even if it is `cf_units only`. Maybe those should be functions?

Remove the return statements but put everything into `if/elif` statements?